lib/test.es6

```diff
@@ -137,11 +308,43 @@ var Test = {
           return;
         }
 
+        // if running via the 'watch:tests' task, we want to be able to run
+        // (require) our test files repeatedly, so this is a hack to make it
+        // work. we copy each test file to a temp filename and load that
+        // instead of the original to avoid getting cached.
+        files = files.map(function(f) {
```

Comment:
Hey @chetan. Was running your branch against my apps test, and found that this behavior here causes the tests to break. Specifically, when running files from a temp directory, all relative `require` statements within those tests files will no longer resolve (i.e., `var EternalJS = require("../app/eternaljs/eternaljs.es6");`. This is a problem because my tests rely on a dependency in order to function properly, and that dependency isn't installed globally.

I think we need to find a different solution to the caching issue (though I appreciate your comment; it helped a lot). Either we write temp files to the associated test file's directory, which is messy especially if Ctrl+C is pressed while the test is running; or we find some other solution. To clarify: Mocha runs files using require rather than through other means?